EIPS/eip-2315.md

```diff
@@ -13,55 +13,67 @@ requires: 3540, 3670, 4200
+Unlike these machines, the Ethereum Virtual Machine _does not_ provide subroutine operations.  Instead, they must be synthesized using the dynamic `JUMP` instruction, which takes its destination on the stack. Further, the EVM provides _only_ dynamic jumps, impeding the static analysis we need.
 
-Unlike these machines, the Ethereum Virtual Machine _does not_ provide subroutine operations.  Instead, they must be synthesized using the dynamic `JUMP` instruction, which takes its destination on the stack. Further, the EVM provides _only_ dynamic jumps, preventing the static analysis we need.
+### We need safe control-flow.
 
-### We want a safer control-flow facilty.
+The EVM has unusually high requirements for safety.  Not only do many smart contracts control inordinately large amounts of valuable Ether, but once placed on the blockchain any defects are visible to attackers and cannot be repaired.  In valid code is an  We need to statically validate important safety constraints code at initialization time.
 
-The EVM has unusually high requirements for safety.  Not only do many smart contracts control inordinately large amounts of valuable Ether, but once placed on the blockchain any defects are visible to attackers and cannot be repaired.  We propose here to statically validate, at initialization time, that code will not execute invalid instructions, jump to invalid locations, or underflow stack.  It will maintain consistent numbers of inputs and outputs for subroutines, and stack growth is bounded in the absence of recursion.
+### We need static control-flow.
 
-### We want an efficient control-flow facility.
+The EVM's lone dynamic jumps cause two major problems. First, the need to synthesize static jumps and subroutines with dynamic jumps uses space and gas with needlessly complex code, as we will show below.
 
-The presence of dynamic jumps and lack of subroutines and static jumps causes two major ineffiencies. First, the need to synthesize static jumps and subroutines with dynamic jumps wastes space and gas and causes needless complexity, as we will show below. 
+Worse, jumps that can dynamically branch to any destination in the code can cause quadratic "path explosions" when traversing the flow of control. For Ethereum, this is a denial-of-service vulnerability that prevents us, at initialization time, from validating the safe use of EVM code and from compiling EVM code to machine code.
 
-Worse, dynamic jumps can branch to any destination in the code, so quadratic "path explosions" are possible when traversing the flow of control. For Ethereum, this is a denial-of-service vulnerability that prevents us, at initialization time, from validating the safe use of EVM code and from compiling EVM code to machine code.
+### We need efficient control-flow.
```

Comment:
```suggestion
### The control-flow needs to be static
```
```suggestion
###  The control-flow needs to be efficient
```

Optional, just to see if this way puts more emphasize on the characteristic this EIP intend to achieve

Reply:
Same issue.  As section titles even the "We need" could be dropped, but it emphasizes that these are motivations.